code/parachain/frame/staking-rewards/src/test/mod.rs

```diff
@@ -131,92 +152,119 @@ fn stake_in_case_of_zero_inflation_should_work() {
-		let total_rewards = 100;
-		let total_shares = 200;
 
 		let staked_asset_id = StakingRewards::pools(PICA::ID).expect("asset_id expected").asset_id;
-		mint_assets([staker], [staked_asset_id], amount * 2);
-		update_total_rewards_and_total_shares_in_rewards_pool(pool_id, total_rewards, total_shares);
+		mint_assets([ALICE], [staked_asset_id], AMOUNT * 2);
+		update_total_rewards_and_total_shares_in_rewards_pool(
+			PICA::ID,
+			TOTAL_REWARDS,
+			TOTAL_SHARES,
+		);
 
-		assert_ok!(StakingRewards::stake(Origin::signed(staker), pool_id, amount, duration_preset));
-		assert_eq!(StakingRewards::stake_count(), 1);
-		let rewards_pool = StakingRewards::pools(pool_id).expect("rewards_pool expected");
-		let reward_multiplier = StakingRewards::reward_multiplier(&rewards_pool, duration_preset)
+		assert_ok!(StakingRewards::stake(Origin::signed(ALICE), PICA::ID, AMOUNT, DURATION_PRESET));
+		let rewards_pool = StakingRewards::pools(PICA::ID).expect("rewards_pool expected");
+		let reward_multiplier = StakingRewards::reward_multiplier(&rewards_pool, DURATION_PRESET)
 			.expect("reward_multiplier expected");
-		let inflation = StakingRewards::boosted_amount(reward_multiplier, amount) * total_rewards /
-			total_shares;
+		let inflation = StakingRewards::boosted_amount(reward_multiplier, AMOUNT) * TOTAL_REWARDS /
+			TOTAL_SHARES;
 		assert_eq!(inflation, 502);
-		let reductions = Reductions::try_from(
-			rewards_pool
-				.rewards
-				.into_inner()
-				.iter()
-				.map(|(asset_id, _reward)| (*asset_id, inflation))
-				.collect::<BTreeMap<_, _>>(),
-		)
-		.expect("reductions expected");
+		let reductions = rewards_pool
+			.rewards
+			.into_inner()
+			.iter()
+			.map(|(asset_id, _reward)| (*asset_id, inflation))
+			.try_collect()
+			.expect("reductions expected");
+
 		assert_eq!(
-			StakingRewards::stakes(StakingRewards::stake_count()),
+			StakingRewards::stakes(1, 0),
 			Some(Stake {
-				owner: staker,
-				reward_pool_id: pool_id,
-				stake: amount,
-				share: StakingRewards::boosted_amount(reward_multiplier, amount),
+				reward_pool_id: PICA::ID,
+				stake: AMOUNT,
+				share: StakingRewards::boosted_amount(reward_multiplier, AMOUNT),
 				reductions,
 				lock: Lock {
 					started_at: <Test as crate::Config>::UnixTime::now(),
-					duration: duration_preset,
+					duration: DURATION_PRESET,
 					unlock_penalty: rewards_pool.lock.unlock_penalty,
 				},
+				financial_nft_id: 0,
 			})
 		);
-		assert_eq!(balance(staked_asset_id, &staker), amount);
-		assert_eq!(balance(staked_asset_id, &StakingRewards::pool_account_id(&pool_id)), amount);
-		assert_last_event::<Test, _>(|e| {
-			matches!(
-				e.event,
-				Event::StakingRewards(crate::Event::Staked { pool_id, owner, amount, position_id, ..})
-					if owner == staker && pool_id == 1 && amount == 100_500u32.into() && position_id == 1
-			)
+
+		assert_eq!(balance(staked_asset_id, &ALICE), AMOUNT);
+		assert_eq!(balance(staked_asset_id, &fnft_asset_account), AMOUNT);
+
```

Comment:
It would be nice if we could also assert the fNFT pallet state after this. i.e check if the fNFT exists

Reply:
28f9c5b5190091deb8ca990a3e8af300c6b5414d